packages/hulypulse-client/src/client.ts

```diff
@@ -0,0 +1,383 @@
+//
+// Copyright © 2025 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+const WS_CLOSE_NORMAL = 1000
+
+type Callback = (
+  key: string,
+  data: JSONValue,
+  info: { msg: JSONObject; subscribed_key: string; run_index: number }
+) => void
+
+const WEBSOCKET_MESSAGE_CONSTANT_PING_REQUEST = 'ping'
+const WEBSOCKET_MESSAGE_CONSTANT_PONG_RESPONSE = 'pong'
+
+type JSONValue = string | number | boolean | null | JSONValue[] | { [key: string]: JSONValue }
+interface JSONObject {
+  [key: string]: JSONValue
+}
+
+interface GetFullResult<T> {
+  data: T
+  etag: string
+  expiresAt: number
+}
+
+interface GetMessage {
+  type: 'get'
+  key: string
+  correlation?: string | number
+}
+
+interface PutMessage {
+  type: 'put'
+  key: string
+  data: JSONValue
+  TTL?: number
+  expiresAt?: number
+  ifMatch?: string
+  ifNoneMatch?: string
+  correlation?: string | number
+}
+
+interface DeleteMessage {
+  type: 'delete'
+  key: string
+  ifMatch?: string
+  correlation?: string | number
+}
+
+interface SubscribeMessages {
+  type: 'sub'
+  key: string
+  correlation?: string | number
+}
+
+interface UnsubscribeMessages {
+  type: 'unsub'
+  key: string
+  correlation?: string | number
+}
+
+interface SubscribesList {
+  type: 'list'
+  correlation?: string | number
+}
+
+interface InfoMessage {
+  type: 'info'
+  correlation?: string | number
+}
+
+type ProtocolMessage =
+  | GetMessage
+  | PutMessage
+  | DeleteMessage
+  | SubscribeMessages
+  | UnsubscribeMessages
+  | SubscribesList
+  | InfoMessage
+
+export class HulypulseClient implements Disposable {
+  private ws: WebSocket | null = null
+  private closed = false
+  private reconnectTimeout: any | undefined
+  private readonly RECONNECT_INTERVAL_MS = 1000
+  private subscribes: Record<string, Callback[]> = {}
```

Comment:
I think it should be `Map`, not `Record` because Map is more suitable for this. Record is just an object.